File: Add-Two-Numbers.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode AddTwoNumbers(ListNode l1, ListNode l2) {
        ListNode h1 = l1 ;
        ListNode h2 = l2 ;

        ListNode res = new ListNode();
        ListNode temp = res ;
        int carry = 0 ;
        while(h1 != null || h2 != null || carry != 0 ){
           int sum = carry;

            if (h1 != null) {
                sum += h1.val;
                h1 = h1.next;
            }

            if (h2 != null) {
                sum += h2.val;
                h2 = h2.next;
            }

            carry = sum / 10;

            temp.next = new ListNode(sum % 10);
            temp = temp.next;
        }
        res = res.next;
        return res ; 
    }
}

File: Add-Two-Numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *h1 = l1;
        ListNode *h2 = l2;

        ListNode *result = new ListNode();
        ListNode *temp_result = result;

        int carry = 0 ; 
        while(h1 && h2){
            int output = h1->val + h2->val + carry ; 
            ListNode *new_node = new ListNode(output%10);
            carry = output / 10 ; 
            temp_result->next = new_node;
            temp_result = new_node;
            h1 = h1->next ;
            h2 = h2->next;
        }
        while(h1){
            int output = h1->val + carry ; 
            ListNode *new_node = new ListNode(output%10);
            carry = output / 10 ; 
            temp_result->next = new_node;
            temp_result = new_node;
            h1 = h1->next ;
        }
        while(h2){
            int output = h2->val + carry ; 
            ListNode *new_node = new ListNode(output%10);
            carry = output / 10 ; 
            temp_result->next = new_node;
            temp_result = new_node;
            h2 = h2->next;
        }

        if(carry){
            ListNode *new_node = new ListNode(carry);
            temp_result->next = new_node;
        }

        return result->next; 
    }
};